code/game/gamemodes/freeze/scion_abilities.dm

```diff
@@ -0,0 +1,261 @@
+
+	panel = "Scion Abilities"
+	charge_max = 300
+	clothes_req = 0
+	icon_state = "frostbite"
+
+/obj/effect/proc_holder/spell/targeted/touch/frostbite/cast()
+	if(!check_frosty(src))
+		return 0
+	..()
+
+/obj/item/weapon/melee/touch_attack/frosty/frostbite
+	name = "\improper freezing hand"
+	desc = "My hand is completely frozen."
+
+	icon_state = "frostbite"
+
+	var/base_dmg = 20
+	var/base_temp = -100
+
+/obj/item/weapon/melee/touch_attack/frosty/frostbite/afterattack(mob/living/carbon/human/target, mob/living/carbon/user, proximity)
+	if(!proximity || target == user || !istype(target) || !istype(user) || user.lying || user.handcuffed)
+		return 0
+	if(is_frosty(target))
+		return 0
+	var/affected = 0
+	if(target.dna.check_mutation(COLDRES))
+		target.dna.remove_mutation(COLDRES)
+		affected++
+	if(target.reagents.has_reagent("inaprovaline"))
+		target.reagents.del_reagent("inaprovaline")
+		affected++
+	var/damageToAfflict = base_dmg - ((affected*0.5)*base_dmg) //does no damage if we removed two effects; half damage if we removed one effect; max damage if we removed none
+	target.apply_damage(damageToAfflict, COLD)
+
+	var/temperatureToAfflict = base_temp - ((affected*0.5)*base_dmg)
+	target.bodytemperature = max(target.bodytemperature - temperatureToAfflict, TCMB)
+	//TODO: feedback
+	return 1
+
+/obj/effect/proc_holder/spell/aoe_turf/extinguish
+	name = "Extinguish"
+	desc = "Quenches nearby fires."
+	panel = "Scion Abilities"
+	charge_max = 600
+	clothes_req = 0
+	range = 0
+
+/obj/effect/proc_holder/spell/aoe_turf/extinguish/cast(list/targets)
+	for(var/turf/simulated/T in targets)
+		new /obj/effect/nanofrost_container/frosty(T)
+	//TODO: logging, feedback
+
+/obj/effect/nanofrost_container/frosty //nanofrost does everything we want for this ability, so why not just spawn one?
+	name = "Ball of ice" //TODO: decent fluff
+	desc = "blah blah blah magic blah blah cold"
+	residue_name = "frozen residue"
+	residue_desc = "residue of magic cold ball thing"
+
+/obj/effect/proc_holder/spell/targeted/touch/refreeze
+	name = "Re-Freeze"
+	desc = "Cools and heals an ally."
+	panel = "Scion Abilities"
+	charge_max = 100
+	clothes_req = 0
+	hand_path = "/obj/item/weapon/melee/touch_attack/frosty/re_freeze"
+
+/obj/effect/proc_holder/spell/targeted/touch/refreeze/cast()
+	if(!check_frosty(src))
+		return 0
+	..()
+
+/obj/item/weapon/melee/touch_attack/frosty/refreeze
+	name = "\improper re-freezing touch"
+	desc = "My hand is cool to the touch."
+
+	icon_state = "refreeze"
+
+	var/base_dmg = 30 //because of coldmod, this heals our allies
+	var/temperature_delta = 40
+
+/obj/item/weapon/melee/touch_attack/frosty/refreeze/afterattack(mob/living/carbon/human/target, mob/living/carbon/user, proximity)
+	if(!proximity || target == user || !istype(target) || !istype(user) || user.lying || user.handcuffed)
+		return 0
+	if(!is_frosty(target))
+		return 0
+
+	target.apply_damage(base_dmg/*, COLD*/)
```

Comment:
...but this throws a "bad argument definition" compile error when I uncomment COLD.
what the hell am I doing wrong?